index.bs

```diff
@@ -1370,77 +896,768 @@ What configuration is the device in after it resets?
+    1. If <a>this</a>.{{USBDevice/[[claimedInterface]]}}[|interfaceIndex|] is <code>false</code>,
+        <a>resolve</a> |promise| and abort these steps.
+    1. Perform the necessary platform-specific steps to reliquish exclusive
+        control over |interfaces|[|interfaceIndex|].
+    1. Set <a>this</a>.{{USBDevice/[[selectedAlternateSetting]]}}[|interfaceIndex|] to <code>0</code>.
+    1. Set <a>this</a>.{{USBDevice/[[claimedInterface]]}}[|interfaceIndex|] to <code>false</code>
+        and <a>resolve</a> |promise|.
+</div>
+
+<div algorithm="USBDevice.selectAlternateInterface(interfaceNumber, alternateSetting)">
+    The <dfn for=USBDevice method>selectAlternateInterface(|interfaceNumber|, |alternateSetting|)</dfn> method,
+    when invoked, MUST return a new {{Promise}} |promise| and run the
+    following steps <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
+    1. Let |activeConfiguration| be the result of <a>finding the current configuration</a>
+        with <a>this</a>.
+    1. Let |interfaces| be |activeConfiguration|.{{USBConfiguration/[[interfaces]]}}.
+    1. Let |interfaceIndex| be the result of <a>finding the interface index</a> with
+        |interfaceNumber| and |activeConfiguration|.
+    1. If |interfaceIndex| is equal to <code>-1</code>, <a>reject</a> |promise|
+        with a {{NotFoundError}} and abort these steps.
+    1. If <a>this</a>.{{USBDevice/[[claimedInterface]]}}[|interfaceIndex|
+        is not <code>true</code>, <a>reject</a> |promise| with an
+        {{InvalidStateError}} and abort these steps.
+    1. Let |interface| be |interfaces|[|interfaceIndex|].
+    1. Let |alternateIndex| be the result of <a>finding the alternate index</a> with
+        |alternateSetting| and |interface|.
+    1. If |alternateIndex| is equal to <code>-1</code>, <a>reject</a> |promise|
+        with a {{NotFoundError}} and abort these steps.
+    1. <a>Abort transfers currently scheduled on an interface</a> with |interface|.
+    1. Issue a <code>SET_INTERFACE</code> <a>control transfer</a> with
+        <code>interfaceNumber</code> set to the |interfaceNumber| and
+        <code>alternateSetting</code> set to the |alternateSetting|.
+        If this step fails <a>reject</a> |promise| with a {{NetworkError}} and
+        abort these steps.
+    1. Set <a>this</a>.{{USBDevice/[[selectedAlternateSetting]]}}[|interfaceIndex|]
+        to |alternateSetting|.
+    1. <a>Resolve</a> |promise|.
+</div>
+
+<div algorithm="USBDevice.controlTransferIn(setup, length)">
+    The <dfn for=USBDevice method>controlTransferIn(|setup|, |length|)</dfn> method,
+    when invoked, MUST return a new {{Promise}} |promise| and run the following steps
+    <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
```

Comment:
```suggestion
        and abort these steps if |promise| is rejected.
```

Reply:
Updated in https://github.com/WICG/webusb/pull/212/commits/5aaf696938ed1dc073f8cc8da8c809e9e5019f41.